### ISSUE TITLE
[Question] substring search

### ISSUE THREAD

**Issue by kuzeko** [nphoff/saxpy#6]
Thanks for sharing this.
I'm not an expert, so I would like to ask you, would it be possible to use your code to search for the top-k similar substrings.

For instance, given

~~~python
target = [ sin(x)*2 for x in range(0, 100) ]
query  = [ sin(x)*2 for x in range(2, 5) ]
~~~

to retrieve  the  windows from `target` that are most similar to `query`?

I guess I should use `s.batch_compare(tStrings,qString)` somehow, but I'm not sure how.
Thanks a lot!

**Comment by nphoff** [comment]
Hey @kuzeko 

Thanks for the question. There was a typo in the readme, which I've fixed which would have made this tough.

What I would do in this example is:

```
from saxpy import SAX
import math

query = [ math.sin(x)*2 for x in range(2, 5)]
target = [math.sin(x)*2 for x in range(0, 100)]

s = SAX(3, 7) # 3 letter words, 7 letter alphabet
(queryString, queryIndices) = s.to_letter_rep(query)
(targetStrings, targetIndices) = s.sliding_window(target, 33, 0.6)
comparisonScores = s.batch_compare(targetStrings, queryString)
```

From there, `comparisonScores` will be a list of comparison scores between subsections of the target and the query.

The lowest `comparisonScores` values will map to which sections were the most similar. The index of the lowest values can be used with `targetIndices` to find the position in the original dataset which were the lowest.

For instance when I ran it, `comparisonScores[2]` and `comparisonScores[3]` were both 0.0, and `targetIndices[2]`, and `targetIndices[3]` were respectively `(2,5)` (exactly the same as the query, which is expected) and `(3,6)` which is pretty close. There will be other entries in `comparisonScores` which are 0.0 because the sine function is periodic.

Good luck!
